search-commons/src/main/java/no/unit/nva/search2/ResourceQuery.java

```diff
@@ -228,11 +226,12 @@ private void addPromotedPublications(UserSettingsClient userSettingsClient, Bool
         if (hasContent(promotedPublications)) {
             removeKey(SORT);  // remove sort to avoid messing up "sorting by score"
             for (int i = 0; i < promotedPublications.size(); i++) {
-                bq.should(
-                    QueryBuilders
-                        .matchQuery(ID, promotedPublications.get(i))
-                        .boost(3.14F + promotedPublications.size() - i)
-                );
+                var uuid = fromUri(promotedPublications.get(i)).getLastPathElement();
+                var qb = QueryBuilders
+                    .matchQuery(IDENTIFIER_KEYWORD, uuid)
+                    .boost(3.14F + promotedPublications.size() - i);
+                logger.info(qb.toString());
+                bq.should(qb);
```

Comment:
UUID is probably Sortableidentifier, which have timestamp "injected" in it.

Reply:
That explains UUID too large....